src/app/(profile)/(components)/ProfileCard/ProfileCard.tsx

```diff
@@ -9,20 +9,22 @@ import {
   mapSeniorityLevel,
   mapSpecializationToTitle,
 } from '@/app/(profile)/profile.mappers'
+import PadlockIcon from '@/assets/icons/PadlockIcon'
 import { Avatar } from '@/components/Avatar/Avatar'
 import TechnologiesRenderer from '@/components/renderers/TechnologiesRenderer'
+import { AppRoutes } from '@/utils/routes'
 import classNames from 'classnames/bind'
+import { useSession } from 'next-auth/react'
 import Link from 'next/link'
 import { useMemo } from 'react'
-import { type UrlObject } from 'url'
 import styles from './ProfileCard.module.scss'
 
 interface ProfileCardProps {
   onClick?: (e: React.MouseEvent) => void
   data: ProfileModel
   withStateStatus?: boolean
   searchTerm?: string | null
-  href: string | UrlObject
+  isHiddenName?: boolean
```

Comment:
For what is this prop? You already check `session` for displaying the names. If so you should check the `session` inside parent component and then pass the value inside `ProfileCard` and then do not check the `session` inside `PrfoileCard`

Reply:
bcs we use this component at home page
and Jakub told me that: on the main page we should display full names regardless of whether the user is logged in.
That's why I'm created that prop. Other way to decide when we will hide full names is check the url path but I don't have idea how to do that in server side component.
Now I see, the name is off the mark, I'll extend it. "isHiddenNameWhenUnlogged" will be okay?

Reply:
Aaa okay :) but still consider checking the session inside `ProfileListItem` and pass the value of isHiddenName prop based on the session, e.g: `isHiddenName = {session && session.user ? true : false}`. Thanks to that we'll still have a `ProfileCard` as a dump component :)
The name of prop is clear for me. I was thinking why you use it but now it is clear for me :)